### PR TITLE
feat(curriculum): create javascript events quiz

### LINKED ISSUE #56207: [HACKTOBERFEST 2024] - Events Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Events quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-events`.

**All quiz questions need to relate to one of these topics:**

- Event object
- addEventListener()
- removeEventListener()
- inline event handlers
- change event
- event bubbling/propagation
- event delegation

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by aakarsh-2004**
Checklist:
- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #56207 

I have created the JavaScript Events Quiz which includes all the subtopics which were specified. I researched a lot in these hours and then after careful observations, I was able to create this quiz.

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 1]
```diff
@@ -17,439 +17,468 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+`.innerHTML`
 
 #### --answer--
 
-Placeholder answer
+`.addEventListener()`
+
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:37]:
```suggestion
`.addEventListener()`
```

you have accidentally added an extra line here

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 2]
```diff
@@ -17,439 +17,468 @@ Answer all of the questions below correctly to pass the quiz.
...
+Following code is an example of?
+```js
+const parentList = document.getElementById('parent-list');
+
+parentList.addEventListener('click', function(event) {
+  if (event.target && event.target.nodeName === 'LI') {
+    console.log('List item clicked:', event.target.textContent);
+  }
+});
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:52]:
````suggestion
Following code is an example of?

```js
const parentList = document.getElementById('parent-list');

parentList.addEventListener('click', function(event) {
  if (event.target && event.target.nodeName === 'LI') {
    console.log('List item clicked:', event.target.textContent);
  }
});
```
````

code fences requires to be surrounded by empty lines

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 3]
```diff
@@ -17,439 +17,468 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+Asynchronous function
 
 #### --answer--
 
-Placeholder answer
+Callback function
+
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:92]:
```suggestion
Callback function
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 4]
```diff
@@ -17,439 +17,468 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+The window object
 
 #### --answer--
 
-Placeholder answer
+The element that triggered the event
+
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:115]:
```suggestion
The element that triggered the event
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 5]
```diff
@@ -17,439 +17,468 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+`addEventListener('click', parameter => myFunction(parameter))`
 
 #### --answer--
 
-Placeholder answer
+`addEventListener('click', () => myFunction(parameter))`
+
```
[the remark this answers is not in the document]
Comment (aakarsh-2004) [reply curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:138]: sure, working on these changes!

**Comment by aakarsh-2004** [comment]
I have added all the necessary changes which were requested, please review it! @ilenia-magoni

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 6]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`.addEventListener()`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Following code is an example of?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:42]:
```suggestion
What does this code demonstrate?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 7]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Callback function
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does the event.target property return?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:96]:
```suggestion
What does the `event.target` property return?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 8]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`addEventListener('click', () => myFunction(parameter))`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the best way to capture an input event across all types of input fields?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:140]:
```suggestion
Which event type captures an input event across all types of input fields?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 9]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+Using `keyup`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:144]:
```suggestion
`keyup`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 10]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+Using `keypress`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:148]:
```suggestion
`keypress`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 11]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+Using `keydown`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:152]:
```suggestion
`keydown`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 12]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+Using `input` event
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:156]:
```suggestion
`input`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 13]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the main advantage of using addEventListener() over inline event handlers?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:162]:
```suggestion
What is the main advantage of using `addEventListener()` over inline event handlers?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 14]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+To prevent the default action associated with an event from being executed
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does the term "event propagation" mean?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:206]:
```suggestion
What does the term "event propagation" refer to?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 15]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+Events are copied to a new window
 
 #### --answer--
 
-Placeholder answer
+The way an event travels through the DOM in phases (capturing, target, and bubbling)
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:222]:
```suggestion
Events travel through the DOM in phases
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 16]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Handling click events on dynamically created elements
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is meant by "default action" of an event in JavaScript?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:294]:
```suggestion
What is meant by the "default action" of an event in JavaScript?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 17]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+The browser's predefined behavior that occurs after an event is triggered
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Why is the once option in addEventListener() useful?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:316]:
```suggestion
Why is the once option in `addEventListener()` useful?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 18]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+It improves performance and reduces memory leaks
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does `DOMContentLoaded` event indicate?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:404]:
```suggestion
What does the `DOMContentLoaded` event indicate?
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 19]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+The element that triggered the event
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How do you correctly pass a parameter to an event handler function in JavaScript?
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:118]:
```suggestion
How do you correctly pass an argument to an event handler function in JavaScript?
```
Arguments are passed to function, parameters are in function definition.

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 20]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+`addEventListener('click', myFunction())`
 
 ---
 
-Placeholder distractor 2
+`addEventListener('click', myFunction.parameter)`
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:126]:
```suggestion
`addEventListener('click', myFunction.argument)`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 21]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+`addEventListener('click', parameter => myFunction(parameter))`
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:130]:
```suggestion
`addEventListener('click', argument => myFunction(argument))`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 22]
```diff
@@ -17,439 +17,449 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+`addEventListener('click', () => myFunction(parameter))`
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:134]:
```suggestion
`addEventListener('click', () => myFunction(argument))`
```

**Comment by aakarsh-2004** [comment]
Changes done! @gikf

**Review by naomi-lgbt — APPROVED** [review]

**Review by gikf — APPROVED** [review]

**Comment by gikf** [comment]
Hi @aakarsh-2004. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**